Tests/xcodeprojTests/Objects/Files/PBXGroupTests.swift

```diff
@@ -52,6 +52,26 @@ final class PBXGroupTests: XCTestCase {
         XCTAssertNotNil(childGroup?.parent)
     }
 
+    func test_addVariantGroup_assignParent() {
+        let project = PBXProj(
+            rootObject: nil,
+            objectVersion: 0,
+            archiveVersion: 0,
+            classes: [:],
+            objects: []
+        )
+
+        let group = PBXGroup(children: [],
+                             sourceTree: .group,
+                             name: "group")
+
+        project.add(object: group)
+
+        let childVariantGroup = try? group.addVariantGroup(named: "child_variant_group").first
+
+        XCTAssertNotNil(childVariantGroup?.parent)
```

Comment:
With this only assertion were are not testing most of the logic that is in the method.
- Are all the components being added as children?
- Are they created with the right name and source tree?
- Are all the expected variant groups returned?